packages/executor/src/utils/etherscan.ts

```diff
@@ -48,24 +51,24 @@ export interface EtherscanResponseBody {
 export const RESPONSE_OK = '1'
 
 export const verifyChugSplashConfig = async (
-  hre: HardhatRuntimeEnvironment,
-  configUri: string
+  configUri: string,
+  provider: ethers.providers.Provider,
+  networkName: string
 ) => {
-  const { etherscanApiKey, etherscanApiEndpoints } = await getEtherscanInfo(hre)
+  const { etherscanApiKey, etherscanApiEndpoints } = await getEtherscanInfo(
+    provider as any, // TODO - figure out how to get the types for this to work correctly
```

Comment:
You should be able to do ethers.providers.JsonRpcProvider here

Reply:
So that works for the functions we define. However, when we eventually pass the provider into `retrieveContractBytecode` and `getEtherscanEndpoints`, there's a type error because the required provider type is `EthereumProvider`. I'm unsure how to resolve that without casting to any since we don't control the typing of those functions.

Reply:
Ah I see. Does it break anything if we set the type to be `EthereumProvider` instead?

Reply:
Yeah we make the type `EthereumProvider` everywhere and that works for running the executor from within the CLI (because the provider that's passed in is of type `EthereumProvider`). However, we run into a problem when running the executor as a standalone process because we need to be able to create the provider from a URL. It seems like hardhat only exports an interface for the `EthereumProvider` which is not sufficient for constructing one from scratch.

Reply:
It looks like the EIP1193Bridge does the trick: https://docs.ethers.io/v5/api/experimental/#experimental-eip1193bridge

(The EthereumProvider inherits from the EIP1193Provider)